### ISSUE TITLE
can't write enum with the highest value

### ISSUE THREAD

**Issue by abeimler** [stephenberry/glaze#1268]
Hi, 
I updated glaze from the latest version 2 to the latest version 3 and now my enums didn't work. (test fails)

Seems like glaze didn't like the "latest" enum value when `write_json`, I'm getting an `error_code::array_element_not_found`.
```cpp
      struct to_json<T>
      {
         template <auto Opts, class... Args>
         GLZ_ALWAYS_INLINE static void op(auto&& value, is_context auto&& ctx, Args&&... args) noexcept
         {
            // TODO: Assumes people dont use strings with chars that need to be escaped for their enum names
            // TODO: Could create a pre quoted map for better performance

            const auto index = static_cast<std::underlying_type_t<T>>(value);
            if (size_t(index) >= refl<T>.keys.size()) {
               ctx.error = error_code::array_element_not_found; ///< error happens here
               return;
            }
```

Try to replicate it with your tests and it also fails:
**json_test.cpp**
```cpp
enum class TestData : uint8_t {
   None,
   A,
   B,
   C,
   D,
   ERROR = 0xFF,
};
struct DummyData {
   uint32_t id{0};

   int32_t a{0};

   TestData b{TestData::None};
   TestData c{TestData::None};
   TestData d{TestData::None};
   TestData e{TestData::None};

   int64_t f{0};
};

template <>
struct glz::meta<TestData> {
   using enum TestData;
   static constexpr auto value = enumerate(None, A, B, C, D, ERROR);
};

template <>
struct glz::meta<DummyData> {
   using T = DummyData;
   static constexpr std::string_view name = "DummyData";
   static constexpr auto value = object(&T::id, &T::a, &T::b, &T::c, &T::d, &T::e, &T::f);
};

suite test_data_struct_tests = [] {
   "test_data_struct"_test = [] {
      const std::vector<DummyData> test_data = {
         DummyData{
            .id = 0,
            .a = 0,
            .b = TestData::None,
            .c = TestData::None,
            .d = TestData::None,
            .e = TestData::None,
            .f = 0x00000000,
         },
         DummyData{
            .id = 1,
            .a = 1,
            .b = TestData::A,
            .c = TestData::B,
            .d = TestData::A,
            .e = TestData::B,
            .f = 0xDDDDDDDD,
         },
         DummyData{
            .id = 2,
            .a = 6,
            .b = TestData::A,
            .c = TestData::B,
            .d = TestData::C,
            .e = TestData::D,
            .f = 0xEEEEEEEE,
         },
         DummyData{
            .id = 3,
            .a = -1,
            .b = TestData::ERROR, ///< problem with this value
            .c = TestData::ERROR,
            .d = TestData::ERROR,
            .e = TestData::ERROR,
            .f = 0xFFFFFFFF,
         },
      };
      expect(glz::refl<TestData>.keys.size() == 6);
      std::string s = glz::write_json(test_data).value_or("error");
      expect(s != "error") << s;
      std::vector<DummyData> in_test_data;
      expect(not glz::read_json(in_test_data, s));
   };
};
```

```bash
FAILED "test_data_struct " ...ze/tests/json_test/json_test.cpp:9233
 error
FAILED "test_data_struct " ...ze/tests/json_test/json_test.cpp:9235

FAILED
tests: 418 (417 passed, 1 failed, 0 compile-time)
asserts: 53132 (53130 passed, 2 failed)
```

**Comment by abeimler** [comment]
I think this check makes less sense when I'm using `enumerate` to "stringify" my enums and I don't care about the underlying value.

either it's an int and I need the check for a valid/existing enum or it's an string I need todo something like `auto color = magic_enum::enum_cast<Color>(color_name);
if (color.has_value()) {`

**Comment by stephenberry** [comment]
Thanks so much for reporting this! I had a major lapse in thinking when making this change. I've reverted to the former, correct approach that uses a map and will optimize the sequenced solution in the future.
The fix will be merged from #1269

**Comment by stephenberry** [comment]
Thanks so much for the unit test as well, I've added it to the test suite.

**Comment by stephenberry** [comment]
Fix released in [v3.2.6](https://github.com/stephenberry/glaze/releases/tag/v3.2.6)
